### DISCUSSION TITLE
Dynamic Load Management

### DISCUSSION THREAD

**Discussion by RomainD2** [dzurikmiroslav/esp32-evse#48] [Ideas]
Hi,

Do you plan to add Dynamic Load Management feature ? I would like to make an EVSE that support this protocol in order to limit the charge current in my EV during peak hours.

Thanks, and nice project !

**Comment by dzurikmiroslav** [comment]
Hello,
Yes you can write own script driver, which will be limiting charging current or pause charging process. Currently for example driver can connect to MQTT broker or read digital/analog input. I plan in future script api for websockets and rs-485/modbus.
https://github.com/dzurikmiroslav/esp32-evse/wiki/Lua#evse-module

**Comment by Radwan-Alalili** [comment]
Hey there,

I hope you doing well, I just wanted to know if I connect the charger to the car does it change the power delivered automatically or it needs me to change the current from the touchscreen

**Comment by taHC81** [comment]
I'm changing current via Home Assistant (using MQTT), based on rules.

**Comment by vitalyxxl** [comment]
I have an idea to use another - a common energy meter (ammeter), included in this circuit. And to track the consumption of this circuit taking into account the charging of the car. The idea is this. For example, we have a limitation in this circuit (phase) of 20 amperes. That is, all consumers cannot take current above this value. But some consumers can turn on and off at different times. Conventionally - an electric kettle / microwave oven / boiler . So during their operation, you can reduce the charging current for the car. We will see the inclusion of these consumers, using another (common) meter / ammeter.
This is dynamically adjustable power.
The simplest method, which I have already implemented, is based on an inexpensive PZEM004T device and a UART-RF transmission module HС-12 (transparent receiver-transmitter UART). I have this bundle installed in the distribution box at the entrance to the house. The real range of stable communication at 9600 is about 100 meters. Which is quite enough for installation even in industrial premises. Data is transmitted via the channel and received by the same HC-12 module and processed by a simple device based on the ESP 8266/ESP32 and displayed on the IPS display 1.3 "240 * 240 (ST7789). 
So, you can receive data from such a remote ammeter / meter with the same UART-RF module HC-12, our EVSE ESP32 board will not even know that this is a radio interface - for it, all work will be like with a regular UART port. Well, and further control of the charging power is performed based on the received data. A similar algorithm is implemented in some Chinese chargers. Very popular functionality.
![RF_POWERMETER](https://github.com/user-attachments/assets/52e22f51-c459-4a38-98fd-5af09090c021)

**Comment by dzurikmiroslav** [comment]
Yes this is possible, you can access any serial port from Lua script. In our country we have electromers from electricity distributor with available serial infra port or for additional fee RS485. I not study API, but I think it is modbus :) I think there should also be a modbus register with current & voltage.
![zpa](https://github.com/user-attachments/assets/79756d1d-2631-49c9-a5a8-309e98db1d83)

**Comment by vitalyxxl** [comment]
Excellent!
It would be good if someone (the author) posted a sample of such a script for reading data via UART/RS 485/IR. As an example for modification to your situation
All the best!
